### ISSUE TITLE
TODO: tests for estimators about raising errors on duplicated column names

### ISSUE THREAD

**Issue by david-cortes** [feature-engine/feature_engine#687]
Per comment in PR: https://github.com/feature-engine/feature_engine/pull/686#pullrequestreview-1544746361

After the PR in the link above gets merged, there will be a general check for duplicated columns for which a test was added in `test_dataframe_checks.py`. This new error raised on duplicated column names should be tested also in the invidual transformers in `estimator_checks/*` after the files gets refactored.

**Comment by solegalli** [comment]
Thank you @david-cortes

**Comment by Morgan-Sell** [comment]
hi @solegalli and @david-cortes,

Slowly dipping my toes back into feature-engine.

Is the idea to create a test/check function on dataframe_checks.py and then add the check functions to the parent transformers in the fit() method?

**Comment by solegalli** [comment]
No, I think the idea was to add a generic test to estimator_checks to test all transformers to ensure that they inherit this functionality properly. I am not sure if it is an overkill or not.

**Comment by Morgan-Sell** [comment]
Ok, so as the code base stands, feature-engine transformers raise errors when there are duplicative feature names, correct? This issue is to create a test that the error is being raised when expected.

**Comment by solegalli** [comment]
Correct!

**Comment by sadia901108** [comment]
Hi! I'd like to work on this issue. 
I'll add tests to ensure transformers raise errors 
when X has duplicated column names. 
Is this still available?

**Comment by solegalli** [comment]
Hey @sadia901108 

Thanks for the support. At the moment, I am thinking of a bigger revamp for feature engine to support polars, so before I have a clearer picture on how we can tackle that, I don't think we should proceed with the current issues.

**Comment by Shubhamjkd01** [comment]
Hi @solegalli  I'm new to open source and would love to contribute. Given the narwhals migration is the current priority, is there a specific module or small task within that effort I could help with?
